detection_rules/rule_loader.py

```diff
@@ -238,13 +238,21 @@ def get_production_rules(verbose=False):
     return filter_rules(load_rules(verbose=verbose).values(), 'maturity', 'production')
 
 
-def find_unneeded_defaults(rule):
+@cached
+def get_non_required_defaults_by_type(rule_type: str) -> dict:
+    """Get list of fields which are not required for a specified rule type."""
+    schema = CurrentSchema.get_schema(rule_type)
+    properties = schema['properties']
+    non_required_defaults = {prop: properties[prop].get('default') for prop in properties
```

Comment:
this doesn't properly handle zero values, unlike:
```suggestion
    non_required_defaults = {prop: 'default' in properties[prop] for prop in properties
```